builder/assignmentEnvImageBuilder.go

```diff
@@ -119,7 +134,10 @@ func (assgnEnv *assignmentEnvironment) verifyLanguage() error {
 	return nil
 }
 
-func (assgnEnv *assignmentEnvironment) writeFromBaseImage() {
+// writeFromBaseImage writes the docker instructions to bytes buffer
+// in 'assignmentEnvironmentImageBuilder' instance staring from the base code runner image.
+// Which is then followed by the required language and its dependencies.
+func (assgnEnv *assignmentEnvironmentImageBuilder) writeFromBaseImage() {
```

Comment:
what is the rationale behind naming the functions `writeFromXXX()`?

Reply:
Depending on whether a docker image for given language is present on docker hub, the docker instructions are written either starting from base image or directly from dependencies. Hence two such functions.